clients/fossid-webapp/src/main/kotlin/FossIdRestService.kt

```diff
@@ -30,11 +35,37 @@ import org.ossreviewtoolkit.fossid.api.status.DownloadStatus
 import org.ossreviewtoolkit.fossid.api.status.ScanStatus
 
 import retrofit2.Call
+import retrofit2.Retrofit
+import retrofit2.converter.jackson.JacksonConverterFactory
 import retrofit2.http.Body
 import retrofit2.http.GET
 import retrofit2.http.POST
 
 interface FossIdRestService {
+    companion object {
+        /**
+         * Create a FossID service instance for communicating with a server running at the given [url],
+         * optionally using a pre-built OkHttp [client].
+         */
+        fun create(
+            url: String,
+            client: OkHttpClient? = null
+        ): FossIdRestService {
+            val fossIdClient = (client ?: OkHttpClient()).newBuilder().build()
```

Comment:
The `.newBuilder().build()` construct is a no-op. You only need to call `.newBuilder()` if you need to create a new client with further configuration based on an existing client, which is not the case here. So we should simply use `.apply { if (client != null) client(client) }` in line 60 instead.